src/app/services/register/register.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Observable, throwError } from 'rxjs';
+import { environment } from 'src/environments/environment';
+import { map, catchError } from 'rxjs/operators';
+import { RegisterResponse } from 'src/app/models/register';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class RegisterService {
+
+  constructor( private http: HttpClient) { }
+
+  sendRegister (username: string, password: string, confirmPsw: string): Observable<RegisterResponse> {
+    const headers = new HttpHeaders({
+      'Content-Type': 'application/json',
+      'Access-Control-Allow-Origin': '*'
+    });
+
+    const options = {
+      headers: headers,
+      observe: 'response' as 'body'
+    };
```

Comment:
if youre not using the headers anywhere else, you can initialize it in the options object

Reply:
corrected